contentcuration/contentcuration/models.py

```diff
@@ -737,25 +766,37 @@ class Channel(models.Model):
     @classmethod
     def filter_edit_queryset(cls, queryset, user):
         user_id = not user.is_anonymous() and user.id
-        user_queryset = User.objects.filter(id=user_id)
-        queryset = queryset.annotate(
-            edit=Exists(user_queryset.filter(editable_channels=OuterRef("id"))),
-        )
 
-        return queryset.filter(edit=True)
+        # it won't return anything
+        if not user_id:
+            return queryset.none()
+
+        edit = Exists(User.editable_channels.through.objects.filter(user_id=user_id, channel_id=OuterRef("id")))
+        return queryset.annotate(edit=edit).filter(edit=True)
 
     @classmethod
     def filter_view_queryset(cls, queryset, user):
         user_id = not user.is_anonymous() and user.id
         user_email = not user.is_anonymous() and user.email
-        user_queryset = User.objects.filter(id=user_id)
+
+        if user_id:
+            edit = Exists(User.editable_channels.through.objects.filter(user_id=user_id, channel_id=OuterRef("id")))
+            view = Exists(User.view_only_channels.through.objects.filter(user_id=user_id, channel_id=OuterRef("id")))
+        else:
+            edit = False
```

Comment:
Should this be `Value(False, BooleanField())`?

Reply:
I was using `Value(False)` which works of course, but I believe I noticed (and will check again) that unless it's a string, it wraps these with `Value`, otherwise it would wrap in a `F`. I don't have a problem using `Value(False)` for explicitness!